modules/statistics/AnalyticsAdapter.js

```diff
@@ -1,130 +1,344 @@
+import {
+    TYPE_OPERATIONAL,
+    TYPE_PAGE,
+    TYPE_TRACK,
+    TYPE_UI
+} from '../../service/statistics/AnalyticsEvents';
+import { getLogger } from 'jitsi-meet-logger';
 import RTCBrowserType from '../RTC/RTCBrowserType';
 import Settings from '../settings/Settings';
 
+const MAX_CACHE_SIZE = 100;
+
+// eslist-disable-line no-undef
+const logger = getLogger(__filename);
+
 /**
- * Interface for analytics handlers.
+ * This class provides an API to lib-jitsi-meet and its users for sending
+ * analytics events. It serves as a bridge to different backend implementations
+ * ("analytics handlers") and a cache for events attempted to be sent before
+ * the analytics handlers were enabled.
+ *
+ * The API is designed to be an easy replacement for the previous version of
+ * this adapter, and is meant to be extended with more convenience methods.
+ *
+ *
+ * The API calls are translated to objects with the following structure, which
+ * are then passed to the sendEvent(event) function of the underlying handlers:
+ *
+ * {
+ *    type,
+ *
+ *    action,
+ *    actionSubject,
+ *    actionSubjectId,
+ *    attributes,
+ *    categories,
+ *    containerId,
+ *    containerType,
+ *    name,
+ *    objectId,
+ *    objectType,
+ *    source,
+ *    tags
+ * }
+ *
+ * The 'type' is one of 'operational', 'page', 'track' or 'ui', and some of the
+ * other properties are considered required according to the type.
+ *
+ * For events with type 'page', the required properties are: name.
+ *
+ * For events with type 'operational' and 'ui', the required properties are:
+ * action, actionSubject, source
+ *
+ * For events with type 'page', the required properties are:
+ * action, actionSubject, source, containerType, containerId, objectType,
+ * objectId
  */
-class AnalyticsAbstract {
+class AnalyticsAdapter {
```

Comment:
isn't this format specific to one backend implementation? should it be named something to match that?

Reply:
oh i see...this is the thing representing 'our' format...so did we just decide to use the same format for our generic format?

Reply:
The format is not specific to one backend implementations. This class dictates how backend implementations work and all backend implementations should be updated to use it. 

I don't quite understand the second question -- what is the difference between "our" and the "generic" format?

Reply:
ok, my understanding is this, let me know if i have something wrong:
1) The events are sent as an object
2) The AnalyticsAdapter adapts those objects into a generic format (the one defined in the comments at the top of AnalyticsAdapter)
3) The handlers will adapt the generic format (from AnalyticsAdapter) into whatever they need.

Is that right?

Here I was just noting that the generic format we used in AnalyticsAdapter seems to be the one from one of the backend handlers we'll be using (I can't remember the name) since, from what I remember, that's what the backend format looked like.

Reply:
Yeah, the format is almost the same